Tests/TelemetryProcessors/SamplingTelemetryProcessor.tests.ts

```diff
@@ -0,0 +1,147 @@
+///<reference path="..\..\typings\globals\node\index.d.ts" />
+///<reference path="..\..\typings\globals\mocha\index.d.ts" />
+///<reference path="..\..\typings\globals\sinon\index.d.ts" />
+
+import assert = require("assert");
+import sinon = require("sinon");
+import Client = require("../../Library/Client");
+
+import Sampling = require("../../TelemetryProcessors/SamplingTelemetryProcessor");
+
+describe("TelemetryProcessors/SamplingTelemetryProcessor", () => {
+    var iKey = "Instrumentation-Key-12345-6789A";
+    var name = "name";
+    var value = 3;
+    var mockData = <any>{ baseData: { properties: {} }, baseType: "BaseTestData" };
+    var client = new Client(iKey);
+
+    describe("#samplingTelemetryProcessor()", () => {
+        it("will not send data on 0% sampling", () => {
+            mockData.sampleRate = 0;
+
+            var result = Sampling.samplingTelemetryProcessor(mockData, {correlationContext: null});
+
+            assert.equal(result, false, "data should not pass");
+        });
+
+        it("will send data roughly 1/3 of the time on 33% sampling", () => {
+            var iterations = 1000;
+            var accepted = 0;
+            mockData.sampleRate = 33;
+
+            for (var i=0; i<iterations; i++) {
+                var result = Sampling.samplingTelemetryProcessor(mockData, {correlationContext: null});
+                if (result) accepted++;
+            }
+
+            assert.ok(accepted > (iterations * 0.25), "data should pass more than 25% of the time");
+            assert.ok(accepted < (iterations * 0.45), "data should pass less than 45% the time");
+        });
+
+        it("will send data roughly 1/2 of the time on 50% sampling", () => {
+            var iterations = 1000;
+            var accepted = 0;
+            mockData.sampleRate = 50;
+
+            for (var i=0; i<iterations; i++) {
+                var result = Sampling.samplingTelemetryProcessor(mockData, {correlationContext: null});
+                if (result) accepted++;
+            }
+
+            assert.ok(accepted > (iterations * 0.40), "data should pass more than 40% of the time");
```

Comment:
This test will fail randomly once it a ~2.72846×10^(-8) runs... I guess we can live with it :)